### PR TITLE
IGNITE-22482 Add topmost root to node configuration

### PR DESCRIPTION
**Pull request by valepakh**
https://issues.apache.org/jira/browse/IGNITE-22482

This patch introduces "ignite" configuration root for the node configuration by moving roots to the extensions of the new `NodeConfigurationSchema` root.

### CODE REVIEW

File: docs/_docs/developers-guide/clients/cpp.adoc [hunk 1]
```diff
@@ -147,24 +147,27 @@ Here is how the client connector configuration looks like:
...
-    enabled: true,
-    clientAuth: "require",
-    keyStore: {
-      path: "KEYSTORE_PATH",
-      password: "SSL_STORE_PASS"
-    },
-    trustStore: {
-      path: "TRUSTSTORE_PATH",
-      password: "SSL_STORE_PASS"
+{
```
Comment (ibessonov) [inline docs/_docs/developers-guide/clients/cpp.adoc:150]: Since we're fixing it anyway, we could remove quotations from the example, making it a proper HOCON rather than a JSON. Maybe it's worth doing as a separate JIRA, to make merge conflicts simpler. What do you think?

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/NodeConfig.java [hunk 2]
```diff
@@ -44,27 +44,25 @@ public class NodeConfig {
...
-                + "  },\n"
-                + "  clientConnector.port: {} ,\n"
-                + "  rest: {\n"
+        return "ignite {\n"
+                + "  network {\n"
+                + "    port: {}\n"
+                + "    nodeFinder.netClusterNodes: [ {} ]\n"
+                + "  }\n"
+                + "  clientConnector.port: {}\n"
+                + "  rest {\n"
```
Comment (ibessonov) [inline modules/cli/src/integrationTest/java/org/apache/ignite/internal/NodeConfig.java:53]: Changing the rest of these lines wasn't really necessary, what motivated you to do more changes than you really needed?
It's everywhere in this PR, I bet it can be 3 times smaller if you remove all unnecessary changes

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/RootKey.java [hunk 3]
```diff
@@ -28,7 +28,7 @@
  * @param <T>    Type of the configuration tree described by the root key.
  * @param <VIEWT> Type of the immutable snapshot view associated with the tree.
  */
-public class RootKey<T extends ConfigurationTree<VIEWT, ?>, VIEWT> {
+public class RootKey<T extends ConfigurationTree<? super VIEWT, ?>, VIEWT> {
```
Comment (ibessonov) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/RootKey.java:31]: I think we don't need this change

File: modules/configuration-api/src/main/java/org/apache/ignite/configuration/SuperRootChange.java [hunk 4]
```diff
@@ -17,20 +17,17 @@
...
-
 /**
  * Interface that represent a "change" for the conjunction of all roots in the configuration.
  */
 public interface SuperRootChange {
     /**
      * Returns a root view for the root key.
      */
-    <V> V viewRoot(RootKey<? extends ConfigurationTree<V, ?>, V> rootKey);
+    <V, C, T extends ConfigurationTree<? super V, ? super C>> V viewRoot(RootKey<T, V> rootKey);
```
Comment (ibessonov) [inline modules/configuration-api/src/main/java/org/apache/ignite/configuration/SuperRootChange.java:27]: What is this for?

File: modules/configuration/src/main/java/org/apache/ignite/internal/configuration/ConfigurationRegistry.java [hunk 5]
```diff
@@ -136,7 +136,7 @@ public void initializeConfigurationWith(ConfigurationSource configurationSource)
      * @param <T> Configuration tree type.
      * @return Public configuration tree.
      */
-    public <V, C, T extends ConfigurationTree<V, C>> T getConfiguration(RootKey<T, V> rootKey) {
+    public <V, C, T extends ConfigurationTree<? super V, C>> T getConfiguration(RootKey<T, V> rootKey) {
```
Comment (ibessonov) [inline modules/configuration/src/main/java/org/apache/ignite/internal/configuration/ConfigurationRegistry.java:139]: I also doubt that this is required, as well as all the rest of `? super`, there should be an easier way

File: modules/client-handler/src/main/java/org/apache/ignite/client/handler/configuration/ClientHandlerConfigurationModule.java [hunk 6]
```diff
@@ -35,7 +34,7 @@ public ConfigurationType type() {
     }
 
     @Override
-    public Collection<RootKey<?, ?>> rootKeys() {
-        return Collections.singleton(ClientConnectorConfiguration.KEY);
+    public Collection<Class<?>> schemaExtensions() {
+        return List.of(ClientConnectorExtensionConfigurationSchema.class);
```
Comment (ptupitsyn) [inline modules/client-handler/src/main/java/org/apache/ignite/client/handler/configuration/ClientHandlerConfigurationModule.java:38]: Is it necessary to create an almost-empty `ClientConnectorExtensionConfigurationSchema` class? Can we somehow pass `ClientConnectorConfigurationSchema` directly?

Comment (ptupitsyn) [reply modules/client-handler/src/main/java/org/apache/ignite/client/handler/configuration/ClientHandlerConfigurationModule.java:38]: It just looks like there must be a shortcut to avoid manually creating an empty class, but I might be wrong.

Comment (ptupitsyn) [reply modules/client-handler/src/main/java/org/apache/ignite/client/handler/configuration/ClientHandlerConfigurationModule.java:38]: Ok, let's keep it if there is no easy way to simplify.

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/NodeConfig.java [hunk 7]
```diff
@@ -44,27 +44,25 @@ public class NodeConfig {
...
-        return "{\n"
-                + "  network: {\n"
-                + "    port: {},\n"
-                + "    nodeFinder: {\n"
-                + "      netClusterNodes: [ {} ]\n"
-                + "    },\n"
-                + "  },\n"
-                + "  clientConnector.port: {} ,\n"
-                + "  rest: {\n"
+        return "ignite {\n"
```
Comment (ibessonov) [inline modules/cli/src/integrationTest/java/org/apache/ignite/internal/NodeConfig.java:47]: You forgot to rollback this one

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/ItNodeConfigurationFileTest.java [hunk 8]
```diff
@@ -57,27 +57,27 @@ public void wrongConfigurationFormat(TestInfo testInfo, @WorkDirectory Path work
...
-                + "      enabled: true,\n"
-                + "      clientAuth: none,\n"
-                + "      keyStore: {\n"
-                + "        path: 123\n"
-                + "      }\n"
-                + "    }\n"
-                + "  }\n"
-                + "}";
+                "ignite {\n"
+                        + "  rest: {\n"
```
Comment (ibessonov) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/ItNodeConfigurationFileTest.java:61]: Something's wrong with padding, please make it as it was

**Review by ibessonov — APPROVED** [review]
